src/runtime-rs/crates/hypervisor/src/ch/inner_device.rs

```diff
@@ -396,12 +415,70 @@ impl CloudHypervisorInner {
+                        // TODO: Replace this symlink workaround if a better approach for rootless socket paths appears.
+                        // In rootless mode the virtiofsd.sock lives under the rootless directory,
+                        // and its full path can exceed the 108-byte Unix domain socket limit.
+                        // To ensure the cloud-hypervisor VMM can connect to virtiofsd, create a
+                        // short symlink inside the rootless directory and point the VMM at it.
+                        let mut fs_cfg = FsConfig::try_from(settings)?;
+                        let rootless_symlink_sanbox_jailer_root =
+                            get_rootless_symlink_sandbox_jailer_root(self.id.as_str());
+
+                        create_dir_all_with_inherit_owner(
+                            rootless_symlink_sanbox_jailer_root.as_str(),
+                            0x750,
+                        )
+                        .map_err(|e| {
+                            anyhow!(
+                                "failed to create rootless sharefs symlink jailer root dir: {}",
+                                e
+                            )
+                        })?;
+                        let virtiofsd_name = fs_cfg.socket.file_name().ok_or_else(|| {
+                            anyhow!(
+                                "failed to get virtiofsd socket file name from path: {:?}",
+                                fs_cfg.socket
+                            )
+                        })?;
+                        let virtiofsd_symlink_path =
+                            PathBuf::from(rootless_symlink_sanbox_jailer_root.as_str())
+                                .join(virtiofsd_name);
+
+                        symlink(&fs_cfg.socket, &virtiofsd_symlink_path).map_err(|e| {
+                            anyhow!(
+                                "failed to create symlink for rootless sharefs socket: {}",
+                                e
+                            )
+                        })?;
+
+                        fs_cfg.socket = virtiofsd_symlink_path;
+
+                        fs_cfg
+                    } else {
+                        FsConfig::try_from(settings)?
+                    };
 
                     shared_fs_devices.push(fs_cfg);
                 }
                 DeviceType::Network(net_device) => {
-                    let net_config = NetConfig::try_from(net_device.config)?;
+                    let mut net_config = NetConfig::default();
+                    if let Some(mac) = net_device.config.guest_mac {
+                        net_config.mac = MacAddr { bytes: mac.0 };
+                    } else {
+                        return Err(anyhow!("Missing mac address for network device"));
+                    }
+                    let netns = self.netns.clone().unwrap_or_default();
+                    let _netns_guard = NetnsGuard::new(&netns).context("new netns guard")?;
```

Comment:
Add a comment here: `// we need ensure opening network device happens in netns.`